src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -430,52 +434,72 @@ public void Initialize()
-                        TraceWriter.Error(errorMsg, ex);
-                        _startupLogger?.LogError(0, ex, errorMsg);
+                        try
+                        {
+                            bindingProvider.Initialize();
+                        }
+                        catch (Exception ex)
+                        {
+                            // If we're unable to initialize a binding provider for any reason, log the error
+                            // and continue
+                            string errorMsg = string.Format("Error initializing binding provider '{0}'", bindingProvider.GetType().FullName);
+                            TraceWriter.Error(errorMsg, ex);
+                            _startupLogger?.LogError(0, ex, errorMsg);
+                        }
                     }
+                    TraceWriter.Verbose("Binding providers initialized.");
                 }
+
                 extensionLoader.LoadBuiltinExtensions(usedBindingTypes);
 
                 var directTypes = GetDirectTypes(functionMetadata);
                 extensionLoader.LoadDirectlyReferencedExtensions(directTypes);
                 extensionLoader.LoadCustomExtensions();
+                TraceWriter.Verbose("Extension loading complete.");
 
                 // Now all extensions have been loaded, the metadata is finalized.
                 // There's a single script binding instance that services all extensions.
                 // give that script binding the metadata for all loaded extensions so it can dispatch to them.
-                var generalProvider = ScriptConfig.BindingProviders.OfType<GeneralScriptBindingProvider>().First();
-                var metadataProvider = this.CreateMetadataProvider();
-                generalProvider.CompleteInitialization(metadataProvider);
+                using (metricsLogger.LatencyEvent(MetricEventNames.HostStartupCreateMetadataProviderLatency))
+                {
+                    var generalProvider = ScriptConfig.BindingProviders.OfType<GeneralScriptBindingProvider>().First();
+                    var metadataProvider = this.CreateMetadataProvider();
+                    generalProvider.CompleteInitialization(metadataProvider);
+                    TraceWriter.Verbose("Metadata provider created.");
+                }
 
                 // Do this after we've loaded the custom extensions. That gives an extension an opportunity to plug in their own implementations.
                 if (storageString != null)
                 {
                     var lockManager = (IDistributedLockManager)Services.GetService(typeof(IDistributedLockManager));
                     _blobLeaseManager = PrimaryHostCoordinator.Create(lockManager, TimeSpan.FromSeconds(15), _hostConfig.HostId, InstanceId, TraceWriter, _hostConfig.LoggerFactory);
+                    TraceWriter.Verbose("Primary host coordinated created.");
```

Comment:
Don't think we need to log this - this is just a ctor call - this isn't where the lock is acquired (that happens on background timer).